<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2024-02-23" />

<title>Integrative Analyses</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Integrative Analyses</h1>
<h4 class="date">February 23, 2024</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                      <span class="at">comment=</span><span class="cn">NA</span>, <span class="at">fig.width =</span> <span class="fl">6.25</span>, <span class="at">fig.height =</span> <span class="dv">5</span>)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(compositions)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">library</span>(tidyverse) </span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">library</span>(mia)</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">library</span>(CCA)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="fu">options</span>(<span class="at">DT.options =</span> <span class="fu">list</span>(</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a>  <span class="at">initComplete =</span> <span class="fu">JS</span>(<span class="st">&quot;function(settings, json) {&quot;</span>,</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a>  <span class="st">&quot;$(this.api().table().header()).css({&#39;background-color&#39;: </span></span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="st">  &#39;#000&#39;, &#39;color&#39;: &#39;#fff&#39;});&quot;</span>,<span class="st">&quot;}&quot;</span>)))</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>base_dir <span class="ot">=</span> <span class="st">&quot;../results&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>sub_dirs <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;outputs&quot;</span>, <span class="st">&quot;figures&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="cf">for</span> (sub_dir <span class="cf">in</span> sub_dirs) {</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  dir_path <span class="ot">=</span> <span class="fu">file.path</span>(base_dir, sub_dir)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(dir_path)) {</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>    <span class="fu">dir.create</span>(dir_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  }</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>lm_eqn <span class="ot">=</span> <span class="cf">function</span>(m){</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  </span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  a <span class="ot">=</span> <span class="fu">unname</span>(<span class="fu">coef</span>(m))[<span class="dv">1</span>]</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  b <span class="ot">=</span> <span class="fu">unname</span>(<span class="fu">coef</span>(m))[<span class="dv">2</span>]</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  p_val <span class="ot">=</span> <span class="fu">summary</span>(m)<span class="sc">$</span>coef[<span class="dv">2</span>, <span class="st">&quot;Pr(&gt;|t|)&quot;</span>]</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  </span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  b <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">sign</span>(b) <span class="sc">&gt;=</span> <span class="dv">0</span>, </span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>             <span class="fu">paste0</span>(<span class="st">&quot; + &quot;</span>, <span class="fu">format</span>(b, <span class="at">digits =</span> <span class="dv">2</span>)), </span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>             <span class="fu">paste0</span>(<span class="st">&quot; - &quot;</span>, <span class="fu">format</span>(<span class="sc">-</span>b, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  </span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  eq <span class="ot">=</span> <span class="fu">substitute</span>(<span class="fu">paste</span>(<span class="fu">italic</span>(y) <span class="sc">==</span> a, b, <span class="fu">italic</span>(x), <span class="st">&quot;, &quot;</span>, <span class="fu">italic</span>(p) <span class="sc">==</span> p_val),</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>                  <span class="fu">list</span>(<span class="at">a =</span> <span class="fu">format</span>(a, <span class="at">digits =</span> <span class="dv">2</span>), <span class="at">b =</span> b,</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>                       <span class="at">p_val =</span> <span class="fu">format</span>(p_val, <span class="at">digits =</span> <span class="dv">2</span>)))</span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  </span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">as.character</span>(<span class="fu">as.expression</span>(eq)))              </span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>}</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a></span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>plot_scatter <span class="ot">=</span> <span class="cf">function</span>(var_name, tax_name, y_lab) {</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>  df_fig <span class="ot">=</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(<span class="fu">c</span>(var_name, tax_name))) <span class="sc">%&gt;%</span></span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">y =</span> var_name) <span class="sc">%&gt;%</span></span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>y, <span class="at">names_to =</span> <span class="st">&quot;tax&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;x&quot;</span>)</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>  </span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>  df_lm <span class="ot">=</span> df_fig <span class="sc">%&gt;%</span> </span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>    <span class="fu">group_by</span>(tax) <span class="sc">%&gt;%</span></span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>    <span class="fu">do</span>(<span class="at">fit =</span> <span class="fu">lm</span>(<span class="at">formula =</span> y <span class="sc">~</span> x, <span class="at">data =</span> .)) <span class="sc">%&gt;%</span></span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">eq =</span> <span class="fu">lm_eqn</span>(<span class="at">m =</span> fit))</span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>  </span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>  df_ann <span class="ot">=</span> df_fig <span class="sc">%&gt;%</span> </span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>    <span class="fu">group_by</span>(tax) <span class="sc">%&gt;%</span></span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">y =</span> <span class="fu">ifelse</span>(<span class="fu">mean</span>(y, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">&gt;</span> <span class="dv">0</span>, </span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>                         <span class="fl">0.5</span> <span class="sc">*</span> <span class="fu">max</span>(y, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>                         <span class="fl">0.2</span> <span class="sc">*</span> <span class="fu">abs</span>(<span class="fu">mean</span>(y, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))),</span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>              <span class="at">x =</span> <span class="fu">median</span>(x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">eq =</span> df_lm<span class="sc">$</span>eq,</span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a>             <span class="at">y_max =</span> <span class="fl">1.05</span> <span class="sc">*</span> y)</span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a>  </span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a>  fig <span class="ot">=</span> df_fig <span class="sc">%&gt;%</span> </span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a>    <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y)) <span class="sc">+</span> </span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>, <span class="at">color =</span> <span class="st">&quot;#BEAED4&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">se =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">&quot;skyblue&quot;</span>, </span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a>                <span class="at">formula =</span> y <span class="sc">~</span> x) <span class="sc">+</span></span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a>    <span class="fu">facet_wrap</span>(.<span class="sc">~</span>tax, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Micorbial abundances&quot;</span>, </span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a>         <span class="at">y =</span> y_lab, </span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a>         <span class="at">title =</span> <span class="cn">NULL</span>) <span class="sc">+</span> </span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a>    <span class="fu">geom_blank</span>(<span class="at">data =</span> df_ann, <span class="fu">aes</span>(<span class="at">y =</span> y_max)) <span class="sc">+</span></span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data =</span> df_ann, <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> eq), <span class="at">size =</span> <span class="dv">3</span>, </span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a>              <span class="at">parse =</span> <span class="cn">TRUE</span>, <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a>    <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">&quot;white&quot;</span>),</span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a>          <span class="at">legend.position =</span> <span class="st">&quot;bottom&quot;</span>,</span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a>          <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a>  <span class="fu">return</span>(fig)</span>
<span id="cb3-55"><a href="#cb3-55" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># Merged data</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>df_v1 <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;../data/df_v1.csv&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>df_v1 <span class="ot">=</span> df_v1 <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">filter</span>(group1 <span class="sc">!=</span> <span class="st">&quot;missing&quot;</span>)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co"># Microbiome data</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>bias_corr_species <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;../results/outputs/bias_corr_species.csv&quot;</span>)</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>bias_corr_species <span class="ot">=</span> bias_corr_species <span class="sc">%&gt;%</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species)) <span class="sc">%&gt;%</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">species =</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species), </span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>                        <span class="fu">paste</span>(<span class="fu">strsplit</span>(species, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">&quot;spp.&quot;</span>),</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>                        <span class="fu">strsplit</span>(species, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])) <span class="sc">%&gt;%</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="fu">ungroup</span>() </span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co"># Significant taxa from ANCOM-BC2</span></span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>sig_species <span class="ot">=</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;../results/outputs/res_ancombc2.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>  <span class="fu">filter</span>(p_val <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>  .<span class="sc">$</span>taxon</span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a><span class="co"># Subset significant taxa</span></span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>df_da_species <span class="ot">=</span>  bias_corr_species <span class="sc">%&gt;%</span></span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a>  <span class="fu">filter</span>(species <span class="sc">%in%</span> sig_species)</span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>df_da_species <span class="ot">=</span> <span class="fu">t</span>(df_da_species)</span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a><span class="fu">colnames</span>(df_da_species) <span class="ot">=</span> df_da_species[<span class="dv">1</span>, ]</span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a>df_da_species <span class="ot">=</span> <span class="fu">data.frame</span>(df_da_species[<span class="sc">-</span><span class="dv">1</span>, , <span class="at">drop =</span> <span class="cn">FALSE</span>], <span class="at">check.names =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;sampleid&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="sc">-</span><span class="dv">1</span>, as.numeric))</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>df_da_species[<span class="fu">is.na</span>(df_da_species)] <span class="ot">=</span> <span class="dv">0</span></span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a></span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>df_add_species <span class="ot">=</span> df_v1 <span class="sc">%&gt;%</span></span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(df_da_species, <span class="at">by =</span> <span class="st">&quot;sampleid&quot;</span>)</span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a>df_add_species <span class="ot">=</span> <span class="fu">data.frame</span>(df_add_species)</span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a>sig_species <span class="ot">=</span> <span class="fu">make.names</span>(sig_species)</span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a></span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a>wb <span class="ot">=</span> <span class="fu">createWorkbook</span>()</span></code></pre></div>
<div id="microbiome-vs.-cytokines" class="section level1">
<h1>Microbiome vs. cytokines</h1>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>t_formula <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="st">&quot;cbind(crp, cd14, cd163) ~ &quot;</span>,</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>                              <span class="fu">paste0</span>(sig_species, <span class="at">collapse =</span> <span class="st">&quot; + &quot;</span>)))</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">manova</span>(t_formula, <span class="at">data =</span> df_add_species)</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>summ <span class="ot">=</span> <span class="fu">summary</span>(fit, <span class="at">test =</span> <span class="st">&quot;Pillai&quot;</span>)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>summ</span></code></pre></div>
<pre><code>                         Df   Pillai approx F num Df den Df  Pr(&gt;F)  
A.muciniphila             1 0.008356   0.5955      3    212 0.61861  
A.onderdonkii             1 0.002446   0.1733      3    212 0.91438  
Anaerovibrio.spp.         1 0.000082   0.0058      3    212 0.99940  
B.adolescentis            1 0.006561   0.4667      3    212 0.70582  
B.caccae                  1 0.003810   0.2703      3    212 0.84677  
B.fragilis                1 0.017932   1.2903      3    212 0.27870  
B.uniformis               1 0.018970   1.3665      3    212 0.25402  
Bacteroides.spp.          1 0.003749   0.2659      3    212 0.84993  
Bifidobacterium.spp.      1 0.019000   1.3687      3    212 0.25333  
Butyricimonas.spp.        1 0.016086   1.1553      3    212 0.32783  
C.celatum                 1 0.012089   0.8648      3    212 0.46019  
Dehalobacterium.spp.      1 0.008515   0.6069      3    212 0.61119  
Lachnobacterium.spp.      1 0.010695   0.7639      3    212 0.51543  
Lachnospira.spp.          1 0.043577   3.2197      3    212 0.02368 *
Megasphaera.spp.          1 0.007058   0.5023      3    212 0.68109  
Methanobrevibacter.spp.   1 0.009872   0.7046      3    212 0.55024  
Odoribacter.spp.          1 0.005665   0.4026      3    212 0.75125  
Paraprevotella.spp.       1 0.011831   0.8461      3    212 0.47005  
RFN20.spp.                1 0.038398   2.8218      3    212 0.03984 *
Succinivibrio.spp.        1 0.026205   1.9016      3    212 0.13035  
Residuals               214                                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>df_summ <span class="ot">=</span> <span class="fu">data.frame</span>(summ<span class="sc">$</span>stats) <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Taxon&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">transmute</span>(Taxon,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                   <span class="at">approx.F =</span> <span class="fu">signif</span>(approx.F, <span class="dv">2</span>),</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>                   num.Df, den.Df, </span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                   <span class="at">P =</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">Pr..F.</span><span class="st">`</span>, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Taxon <span class="sc">!=</span> <span class="st">&quot;Residuals&quot;</span>)</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="fu">addWorksheet</span>(wb, <span class="st">&quot;cyto_species&quot;</span>)</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="fu">writeData</span>(wb, <span class="st">&quot;cyto_species&quot;</span>, df_summ)</span></code></pre></div>
</div>
<div id="microbiome-vs.-scfas" class="section level1">
<h1>Microbiome vs. SCFAs</h1>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>t_formula <span class="ot">=</span> <span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="st">&quot;cbind(acetate, valerate) ~ &quot;</span>,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>                              <span class="fu">paste0</span>(sig_species, <span class="at">collapse =</span> <span class="st">&quot; + &quot;</span>)))</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">manova</span>(t_formula, <span class="at">data =</span> df_add_species)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>summ <span class="ot">=</span> <span class="fu">summary</span>(fit, <span class="at">test =</span> <span class="st">&quot;Pillai&quot;</span>)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>summ</span></code></pre></div>
<pre><code>                         Df   Pillai approx F num Df den Df  Pr(&gt;F)   
A.muciniphila             1 0.004696   0.4624      2    196 0.63047   
A.onderdonkii             1 0.002037   0.2000      2    196 0.81890   
Anaerovibrio.spp.         1 0.007019   0.6927      2    196 0.50143   
B.adolescentis            1 0.001395   0.1369      2    196 0.87218   
B.caccae                  1 0.009612   0.9512      2    196 0.38807   
B.fragilis                1 0.020457   2.0467      2    196 0.13192   
B.uniformis               1 0.060359   6.2952      2    196 0.00224 **
Bacteroides.spp.          1 0.012146   1.2049      2    196 0.30192   
Bifidobacterium.spp.      1 0.001698   0.1667      2    196 0.84656   
Butyricimonas.spp.        1 0.024454   2.4566      2    196 0.08836 . 
C.celatum                 1 0.009003   0.8903      2    196 0.41220   
Dehalobacterium.spp.      1 0.015251   1.5177      2    196 0.22178   
Lachnobacterium.spp.      1 0.001698   0.1667      2    196 0.84655   
Lachnospira.spp.          1 0.003468   0.3410      2    196 0.71147   
Megasphaera.spp.          1 0.035376   3.5940      2    196 0.02931 * 
Methanobrevibacter.spp.   1 0.017202   1.7153      2    196 0.18261   
Odoribacter.spp.          1 0.001740   0.1708      2    196 0.84310   
Paraprevotella.spp.       1 0.023803   2.3896      2    196 0.09433 . 
RFN20.spp.                1 0.016776   1.6721      2    196 0.19051   
Succinivibrio.spp.        1 0.033204   3.3658      2    196 0.03654 * 
Residuals               197                                           
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>df_summ <span class="ot">=</span> <span class="fu">data.frame</span>(summ<span class="sc">$</span>stats) <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Taxon&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">transmute</span>(Taxon,</span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>                   <span class="at">approx.F =</span> <span class="fu">signif</span>(approx.F, <span class="dv">2</span>),</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>                   num.Df, den.Df, </span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>                   <span class="at">P =</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">Pr..F.</span><span class="st">`</span>, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Taxon <span class="sc">!=</span> <span class="st">&quot;Residuals&quot;</span>)</span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="fu">addWorksheet</span>(wb, <span class="st">&quot;scfa_species&quot;</span>)</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="fu">writeData</span>(wb, <span class="st">&quot;scfa_species&quot;</span>, df_summ)</span></code></pre></div>
</div>
<div id="cytokines-vs.-scfas" class="section level1">
<h1>Cytokines vs. SCFAs</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">manova</span>(<span class="fu">cbind</span>(crp, cd14, cd163) <span class="sc">~</span> </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>               acetate <span class="sc">+</span> valerate, </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>             <span class="at">data =</span> df_v1)</span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>summ <span class="ot">=</span> <span class="fu">summary</span>(fit, <span class="at">test =</span> <span class="st">&quot;Pillai&quot;</span>)</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>summ</span></code></pre></div>
<pre><code>           Df   Pillai approx F num Df den Df  Pr(&gt;F)  
acetate     1 0.033928   2.5286      3    216 0.05826 .
valerate    1 0.016705   1.2232      3    216 0.30217  
Residuals 218                                          
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>df_summ <span class="ot">=</span> <span class="fu">data.frame</span>(summ<span class="sc">$</span>stats) <span class="sc">%&gt;%</span></span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Taxon&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">transmute</span>(Taxon,</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>                   <span class="at">approx.F =</span> <span class="fu">signif</span>(approx.F, <span class="dv">2</span>),</span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a>                   num.Df, den.Df, </span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a>                   <span class="at">P =</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">Pr..F.</span><span class="st">`</span>, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Taxon <span class="sc">!=</span> <span class="st">&quot;Residuals&quot;</span>)</span>
<span id="cb13-8"><a href="#cb13-8" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" tabindex="-1"></a><span class="fu">addWorksheet</span>(wb, <span class="st">&quot;cyto_scfa&quot;</span>)</span>
<span id="cb13-10"><a href="#cb13-10" tabindex="-1"></a><span class="fu">writeData</span>(wb, <span class="st">&quot;cyto_scfa&quot;</span>, df_summ)</span></code></pre></div>
</div>
<div id="outputs" class="section level1">
<h1>Outputs</h1>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">saveWorkbook</span>(wb, <span class="at">file =</span> <span class="st">&quot;../results/outputs/res_integrate.xlsx&quot;</span>, <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.0

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] splines   stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] DT_0.30                         CCA_1.2.2                      
 [3] fields_15.2                     viridisLite_0.4.2              
 [5] spam_2.10-0                     fda_6.1.4                      
 [7] deSolve_1.40                    fds_1.8                        
 [9] RCurl_1.98-1.12                 rainbow_3.8                    
[11] pcaPP_2.0-4                     MASS_7.3-60                    
[13] ggpubr_0.6.0                    mia_1.10.0                     
[15] MultiAssayExperiment_1.28.0     TreeSummarizedExperiment_2.10.0
[17] Biostrings_2.70.1               XVector_0.42.0                 
[19] SingleCellExperiment_1.24.0     SummarizedExperiment_1.32.0    
[21] Biobase_2.62.0                  GenomicRanges_1.54.1           
[23] GenomeInfoDb_1.38.0             IRanges_2.36.0                 
[25] S4Vectors_0.40.1                BiocGenerics_0.48.0            
[27] MatrixGenerics_1.14.0           matrixStats_1.0.0              
[29] lubridate_1.9.3                 forcats_1.0.0                  
[31] stringr_1.5.0                   dplyr_1.1.3                    
[33] purrr_1.0.2                     tidyr_1.3.0                    
[35] tibble_3.2.1                    ggplot2_3.4.4                  
[37] tidyverse_2.0.0                 compositions_2.0-6             
[39] openxlsx_4.2.5.2                readr_2.1.4                    

loaded via a namespace (and not attached):
  [1] tensorA_0.36.2              rstudioapi_0.15.0          
  [3] jsonlite_1.8.7              magrittr_2.0.3             
  [5] ggbeeswarm_0.7.2            rmarkdown_2.25             
  [7] fs_1.6.3                    zlibbioc_1.48.0            
  [9] vctrs_0.6.4                 memoise_2.0.1              
 [11] DelayedMatrixStats_1.24.0   rstatix_0.7.2              
 [13] htmltools_0.5.6.1           S4Arrays_1.2.0             
 [15] broom_1.0.5                 BiocNeighbors_1.20.0       
 [17] SparseArray_1.2.2           hdrcde_3.4                 
 [19] pracma_2.4.4                sass_0.4.7                 
 [21] KernSmooth_2.23-21          bslib_0.5.1                
 [23] htmlwidgets_1.6.2           plyr_1.8.9                 
 [25] DECIPHER_2.30.0             cachem_1.0.8               
 [27] igraph_1.5.1                lifecycle_1.0.3            
 [29] pkgconfig_2.0.3             rsvd_1.0.5                 
 [31] Matrix_1.6-1.1              R6_2.5.1                   
 [33] fastmap_1.1.1               GenomeInfoDbData_1.2.11    
 [35] digest_0.6.33               colorspace_2.1-0           
 [37] scater_1.30.0               irlba_2.3.5.1              
 [39] RSQLite_2.3.2               vegan_2.6-4                
 [41] beachmat_2.18.0             fansi_1.0.5                
 [43] timechange_0.2.0            abind_1.4-5                
 [45] mgcv_1.8-42                 compiler_4.3.1             
 [47] bit64_4.0.5                 withr_2.5.1                
 [49] backports_1.4.1             BiocParallel_1.36.0        
 [51] carData_3.0-5               viridis_0.6.4              
 [53] DBI_1.1.3                   maps_3.4.2                 
 [55] ggsignif_0.6.4              bayesm_3.1-6               
 [57] DelayedArray_0.28.0         bluster_1.12.0             
 [59] permute_0.9-7               tools_4.3.1                
 [61] vipor_0.4.5                 beeswarm_0.4.0             
 [63] ape_5.7-1                   zip_2.3.0                  
 [65] glue_1.6.2                  nlme_3.1-162               
 [67] grid_4.3.1                  cluster_2.1.4              
 [69] reshape2_1.4.4              generics_0.1.3             
 [71] gtable_0.3.4                tzdb_0.4.0                 
 [73] hms_1.1.3                   car_3.1-2                  
 [75] BiocSingular_1.18.0         ScaledMatrix_1.10.0        
 [77] utf8_1.2.4                  ggrepel_0.9.4              
 [79] pillar_1.9.0                vroom_1.6.4                
 [81] yulab.utils_0.1.0           robustbase_0.99-0          
 [83] treeio_1.26.0               lattice_0.21-8             
 [85] bit_4.0.5                   ks_1.14.2                  
 [87] tidyselect_1.2.0            DirichletMultinomial_1.44.0
 [89] scuttle_1.12.0              knitr_1.45                 
 [91] gridExtra_2.3               xfun_0.40                  
 [93] DEoptimR_1.1-3              stringi_1.7.12             
 [95] lazyeval_0.2.2              yaml_2.3.7                 
 [97] evaluate_0.22               codetools_0.2-19           
 [99] cli_3.6.1                   munsell_0.5.0              
[101] jquerylib_0.1.4             Rcpp_1.0.11                
[103] parallel_4.3.1              blob_1.2.4                 
[105] dotCall64_1.1-1             mclust_6.0.1               
[107] sparseMatrixStats_1.14.0    bitops_1.0-7               
[109] mvtnorm_1.2-3               decontam_1.22.0            
[111] tidytree_0.4.5              scales_1.2.1               
[113] crayon_1.5.2                rlang_1.1.1                </code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
