<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />


<meta name="date" content="2024-08-09" />

<title>Microbial Correlation Analyses</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Microbial Correlation Analyses</h1>
<h4 class="date">August 09, 2024</h4>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># rm(list = ls())</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(<span class="at">echo=</span><span class="cn">TRUE</span>, <span class="at">warning=</span><span class="cn">FALSE</span>, <span class="at">message=</span><span class="cn">FALSE</span>, </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                      <span class="at">comment=</span><span class="cn">NA</span>, <span class="at">fig.width =</span> <span class="fl">6.25</span>, <span class="at">fig.height =</span> <span class="dv">5</span>)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(tidyverse) </span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">library</span>(igraph)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">library</span>(ggraph)</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(tidygraph)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">library</span>(microbiome)</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="fu">library</span>(ANCOMBC)</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="fu">options</span>(<span class="at">DT.options =</span> <span class="fu">list</span>(</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a>  <span class="at">initComplete =</span> <span class="fu">JS</span>(<span class="st">&quot;function(settings, json) {&quot;</span>,</span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a>  <span class="st">&quot;$(this.api().table().header()).css({&#39;background-color&#39;: </span></span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="st">  &#39;#000&#39;, &#39;color&#39;: &#39;#fff&#39;});&quot;</span>,<span class="st">&quot;}&quot;</span>)))</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>base_dir <span class="ot">=</span> <span class="st">&quot;../results&quot;</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>sub_dirs <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;outputs&quot;</span>, <span class="st">&quot;figures&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="cf">for</span> (sub_dir <span class="cf">in</span> sub_dirs) {</span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  dir_path <span class="ot">=</span> <span class="fu">file.path</span>(base_dir, sub_dir)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>  <span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(dir_path)) {</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>    <span class="fu">dir.create</span>(dir_path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  }</span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>get_upper_tri <span class="ot">=</span> <span class="cf">function</span>(cormat){</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>    cormat[<span class="fu">lower.tri</span>(cormat)] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>    <span class="fu">diag</span>(cormat) <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>    <span class="fu">return</span>(cormat)</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>}</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>data_preprocess <span class="ot">=</span> <span class="cf">function</span>(res, <span class="at">type =</span> <span class="st">&quot;linear&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, tax) {</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="cf">if</span> (type <span class="sc">==</span> <span class="st">&quot;linear&quot;</span>) {</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>    df_corr <span class="ot">=</span> res<span class="sc">$</span>corr_fl</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>    df_corr <span class="ot">=</span> res<span class="sc">$</span>dcorr_fl</span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>  }</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a>  </span>
<span id="cb3-14"><a href="#cb3-14" tabindex="-1"></a>  <span class="cf">if</span> (level <span class="sc">==</span> <span class="st">&quot;Species&quot;</span>) {</span>
<span id="cb3-15"><a href="#cb3-15" tabindex="-1"></a>    tax_name <span class="ot">=</span> <span class="fu">colnames</span>(df_corr)</span>
<span id="cb3-16"><a href="#cb3-16" tabindex="-1"></a>    tax_name <span class="ot">=</span> <span class="fu">sapply</span>(tax_name, <span class="cf">function</span>(x) {</span>
<span id="cb3-17"><a href="#cb3-17" tabindex="-1"></a>      name <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, x), </span>
<span id="cb3-18"><a href="#cb3-18" tabindex="-1"></a>             <span class="fu">paste</span>(<span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">&quot;spp.&quot;</span>),</span>
<span id="cb3-19"><a href="#cb3-19" tabindex="-1"></a>             <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, x), </span>
<span id="cb3-20"><a href="#cb3-20" tabindex="-1"></a>                    <span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], x))</span>
<span id="cb3-21"><a href="#cb3-21" tabindex="-1"></a>      <span class="fu">return</span>(name)</span>
<span id="cb3-22"><a href="#cb3-22" tabindex="-1"></a>    })</span>
<span id="cb3-23"><a href="#cb3-23" tabindex="-1"></a>    <span class="fu">colnames</span>(df_corr) <span class="ot">=</span> tax_name</span>
<span id="cb3-24"><a href="#cb3-24" tabindex="-1"></a>    <span class="fu">rownames</span>(df_corr) <span class="ot">=</span> tax_name</span>
<span id="cb3-25"><a href="#cb3-25" tabindex="-1"></a>    </span>
<span id="cb3-26"><a href="#cb3-26" tabindex="-1"></a>    df_corr <span class="ot">=</span> df_corr[tax, tax]</span>
<span id="cb3-27"><a href="#cb3-27" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb3-28"><a href="#cb3-28" tabindex="-1"></a>    tax_name <span class="ot">=</span> <span class="fu">colnames</span>(df_corr)</span>
<span id="cb3-29"><a href="#cb3-29" tabindex="-1"></a>    tax_name <span class="ot">=</span> <span class="fu">sapply</span>(tax_name, <span class="cf">function</span>(x) </span>
<span id="cb3-30"><a href="#cb3-30" tabindex="-1"></a>      <span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</span>
<span id="cb3-31"><a href="#cb3-31" tabindex="-1"></a>    <span class="fu">colnames</span>(df_corr) <span class="ot">=</span> tax_name</span>
<span id="cb3-32"><a href="#cb3-32" tabindex="-1"></a>    <span class="fu">rownames</span>(df_corr) <span class="ot">=</span> tax_name</span>
<span id="cb3-33"><a href="#cb3-33" tabindex="-1"></a>    </span>
<span id="cb3-34"><a href="#cb3-34" tabindex="-1"></a>    df_corr <span class="ot">=</span> df_corr[tax, tax]</span>
<span id="cb3-35"><a href="#cb3-35" tabindex="-1"></a>  }</span>
<span id="cb3-36"><a href="#cb3-36" tabindex="-1"></a>  </span>
<span id="cb3-37"><a href="#cb3-37" tabindex="-1"></a>  tax_name <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">colnames</span>(df_corr))</span>
<span id="cb3-38"><a href="#cb3-38" tabindex="-1"></a>  df_corr <span class="ot">=</span> df_corr[tax_name, tax_name]</span>
<span id="cb3-39"><a href="#cb3-39" tabindex="-1"></a>  </span>
<span id="cb3-40"><a href="#cb3-40" tabindex="-1"></a>  df_clean <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">get_upper_tri</span>(df_corr)) <span class="sc">%&gt;%</span></span>
<span id="cb3-41"><a href="#cb3-41" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;var1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-42"><a href="#cb3-42" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="sc">-</span>var1, <span class="at">names_to =</span> <span class="st">&quot;var2&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>) </span>
<span id="cb3-43"><a href="#cb3-43" tabindex="-1"></a>  </span>
<span id="cb3-44"><a href="#cb3-44" tabindex="-1"></a>  <span class="co"># Correct for names</span></span>
<span id="cb3-45"><a href="#cb3-45" tabindex="-1"></a>  df_name <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">var1 =</span> <span class="fu">unique</span>(df_clean<span class="sc">$</span>var1),</span>
<span id="cb3-46"><a href="#cb3-46" tabindex="-1"></a>                       <span class="at">var2 =</span> <span class="fu">unique</span>(df_clean<span class="sc">$</span>var2))</span>
<span id="cb3-47"><a href="#cb3-47" tabindex="-1"></a>  </span>
<span id="cb3-48"><a href="#cb3-48" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_len</span>(<span class="fu">nrow</span>(df_name))) {</span>
<span id="cb3-49"><a href="#cb3-49" tabindex="-1"></a>    df_clean<span class="sc">$</span>var2[df_clean<span class="sc">$</span>var2 <span class="sc">==</span> df_name<span class="sc">$</span>var2[i]] <span class="ot">=</span> df_name<span class="sc">$</span>var1[i]</span>
<span id="cb3-50"><a href="#cb3-50" tabindex="-1"></a>  }</span>
<span id="cb3-51"><a href="#cb3-51" tabindex="-1"></a>  df_clean <span class="ot">=</span> df_clean <span class="sc">%&gt;%</span></span>
<span id="cb3-52"><a href="#cb3-52" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(value)) <span class="sc">%&gt;%</span></span>
<span id="cb3-53"><a href="#cb3-53" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">value =</span> <span class="fu">round</span>(value, <span class="dv">2</span>))</span>
<span id="cb3-54"><a href="#cb3-54" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="co"># OTU table</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>otu_table <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">&quot;../data/otu-table.tsv&quot;</span>)</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>otu_id <span class="ot">=</span> otu_table<span class="sc">$</span><span class="st">`</span><span class="at">#OTU ID</span><span class="st">`</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>otu_table <span class="ot">=</span> <span class="fu">data.frame</span>(otu_table[, <span class="sc">-</span><span class="dv">1</span>], <span class="at">check.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> otu_id)</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co"># Taxonomy table</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>tax <span class="ot">=</span> <span class="fu">read_tsv</span>(<span class="st">&quot;../data/taxonomy.tsv&quot;</span>)</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>otu_id <span class="ot">=</span> tax<span class="sc">$</span><span class="st">`</span><span class="at">Feature ID</span><span class="st">`</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>tax <span class="ot">=</span> <span class="fu">data.frame</span>(tax[, <span class="sc">-</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>)], <span class="at">row.names =</span> otu_id)</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>tax <span class="ot">=</span> tax <span class="sc">%&gt;%</span> </span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> Taxon, </span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a>           <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;Kingdom&quot;</span>, <span class="st">&quot;Phylum&quot;</span>, <span class="st">&quot;Class&quot;</span>, <span class="st">&quot;Order&quot;</span>, </span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a>                    <span class="st">&quot;Family&quot;</span>, <span class="st">&quot;Genus&quot;</span>, <span class="st">&quot;Species&quot;</span>),</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a>           <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a>  <span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate_all</span>(<span class="cf">function</span>(x) <span class="fu">strsplit</span>(x, <span class="st">&quot;__&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]) <span class="sc">%&gt;%</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Species =</span> <span class="fu">ifelse</span>(<span class="sc">!</span><span class="fu">is.na</span>(Species) <span class="sc">&amp;</span> <span class="sc">!</span><span class="fu">is.na</span>(Genus),</span>
<span id="cb4-18"><a href="#cb4-18" tabindex="-1"></a>                          <span class="fu">paste</span>(<span class="fu">ifelse</span>(<span class="fu">strsplit</span>(Genus, <span class="st">&quot;&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>] <span class="sc">==</span> <span class="st">&quot;[&quot;</span>,</span>
<span id="cb4-19"><a href="#cb4-19" tabindex="-1"></a>                                       <span class="fu">strsplit</span>(Genus, <span class="st">&quot;&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>],</span>
<span id="cb4-20"><a href="#cb4-20" tabindex="-1"></a>                                       <span class="fu">strsplit</span>(Genus, <span class="st">&quot;&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]), Species, <span class="at">sep =</span> <span class="st">&quot;.&quot;</span>),</span>
<span id="cb4-21"><a href="#cb4-21" tabindex="-1"></a>                          <span class="cn">NA</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-22"><a href="#cb4-22" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb4-23"><a href="#cb4-23" tabindex="-1"></a>tax <span class="ot">=</span> <span class="fu">as.matrix</span>(tax)</span>
<span id="cb4-24"><a href="#cb4-24" tabindex="-1"></a><span class="fu">rownames</span>(tax) <span class="ot">=</span> otu_id</span>
<span id="cb4-25"><a href="#cb4-25" tabindex="-1"></a>tax[tax <span class="sc">==</span> <span class="st">&quot;&quot;</span>] <span class="ot">=</span> <span class="cn">NA</span></span>
<span id="cb4-26"><a href="#cb4-26" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" tabindex="-1"></a><span class="co"># Meta data</span></span>
<span id="cb4-28"><a href="#cb4-28" tabindex="-1"></a>meta_data <span class="ot">=</span> <span class="fu">read_csv</span>(<span class="st">&quot;../data/df_v1.csv&quot;</span>)</span>
<span id="cb4-29"><a href="#cb4-29" tabindex="-1"></a>meta_data<span class="sc">$</span>status <span class="ot">=</span> <span class="fu">factor</span>(meta_data<span class="sc">$</span>status, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;nc&quot;</span>, <span class="st">&quot;sc&quot;</span>))</span>
<span id="cb4-30"><a href="#cb4-30" tabindex="-1"></a>meta_data<span class="sc">$</span>time2aids <span class="ot">=</span> <span class="fu">factor</span>(meta_data<span class="sc">$</span>time2aids,</span>
<span id="cb4-31"><a href="#cb4-31" tabindex="-1"></a>                             <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;never&quot;</span>, <span class="st">&quot;&gt; 10 yrs&quot;</span>,</span>
<span id="cb4-32"><a href="#cb4-32" tabindex="-1"></a>                                        <span class="st">&quot;5 - 10 yrs&quot;</span>, <span class="st">&quot;&lt; 5 yrs&quot;</span>))</span>
<span id="cb4-33"><a href="#cb4-33" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" tabindex="-1"></a><span class="co"># Phyloseq object</span></span>
<span id="cb4-35"><a href="#cb4-35" tabindex="-1"></a>OTU <span class="ot">=</span> <span class="fu">otu_table</span>(otu_table, <span class="at">taxa_are_rows =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-36"><a href="#cb4-36" tabindex="-1"></a>META <span class="ot">=</span> <span class="fu">sample_data</span>(meta_data)</span>
<span id="cb4-37"><a href="#cb4-37" tabindex="-1"></a><span class="fu">sample_names</span>(META) <span class="ot">=</span> meta_data<span class="sc">$</span>sampleid</span>
<span id="cb4-38"><a href="#cb4-38" tabindex="-1"></a>TAX <span class="ot">=</span> <span class="fu">tax_table</span>(tax)</span>
<span id="cb4-39"><a href="#cb4-39" tabindex="-1"></a>otu_data <span class="ot">=</span> <span class="fu">phyloseq</span>(OTU, TAX, META)</span>
<span id="cb4-40"><a href="#cb4-40" tabindex="-1"></a>species_data <span class="ot">=</span> <span class="fu">aggregate_taxa</span>(otu_data, <span class="st">&quot;Species&quot;</span>)</span>
<span id="cb4-41"><a href="#cb4-41" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" tabindex="-1"></a>tse <span class="ot">=</span> mia<span class="sc">::</span><span class="fu">makeTreeSummarizedExperimentFromPhyloseq</span>(otu_data)</span>
<span id="cb4-43"><a href="#cb4-43" tabindex="-1"></a>tse <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">!=</span> <span class="st">&quot;missing&quot;</span>]</span>
<span id="cb4-44"><a href="#cb4-44" tabindex="-1"></a>tse1 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g1&quot;</span>]</span>
<span id="cb4-45"><a href="#cb4-45" tabindex="-1"></a>tse2 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g2&quot;</span>]</span>
<span id="cb4-46"><a href="#cb4-46" tabindex="-1"></a>tse3 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g3&quot;</span>]</span>
<span id="cb4-47"><a href="#cb4-47" tabindex="-1"></a>tse4 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g4&quot;</span>]</span></code></pre></div>
<ol style="list-style-type: decimal">
<li><p>Number of stool samples 243,</p></li>
<li><p>Number of taxa 128.</p></li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>key_species <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;A.muciniphila&quot;</span>, <span class="st">&quot;B.caccae&quot;</span>, <span class="st">&quot;B.fragilis&quot;</span>, <span class="st">&quot;B.uniformis&quot;</span>, </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>                <span class="st">&quot;Bacteroides spp.&quot;</span>, <span class="st">&quot;Butyricimonas spp.&quot;</span>, <span class="st">&quot;Dehalobacterium spp.&quot;</span>, </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>                <span class="st">&quot;Methanobrevibacter spp.&quot;</span>, <span class="st">&quot;Odoribacter spp.&quot;</span>)</span></code></pre></div>
<div id="linear-correlations" class="section level1 tabset">
<h1 class="tabset">Linear correlations</h1>
<div id="run-secom" class="section level2">
<h2>Run SECOM</h2>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>tse1 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g1&quot;</span>]</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>tse2 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g2&quot;</span>]</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>tse3 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g3&quot;</span>]</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>tse4 <span class="ot">=</span> tse[, tse<span class="sc">$</span>group1 <span class="sc">==</span> <span class="st">&quot;g4&quot;</span>]</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>res_linear1 <span class="ot">=</span> <span class="fu">secom_linear</span>(<span class="at">data =</span> <span class="fu">list</span>(tse1), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>                           <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>                           <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>                           <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>, </span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>                           <span class="at">soft =</span> <span class="cn">FALSE</span>, <span class="at">thresh_len =</span> <span class="dv">20</span>, <span class="at">n_cv =</span> <span class="dv">10</span>, </span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>                           <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>res_linear2 <span class="ot">=</span> <span class="fu">secom_linear</span>(<span class="at">data =</span> <span class="fu">list</span>(tse2), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>                           <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>                           <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>                           <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>, </span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>                           <span class="at">soft =</span> <span class="cn">FALSE</span>, <span class="at">thresh_len =</span> <span class="dv">20</span>, <span class="at">n_cv =</span> <span class="dv">10</span>, </span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>                           <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>res_linear3 <span class="ot">=</span> <span class="fu">secom_linear</span>(<span class="at">data =</span> <span class="fu">list</span>(tse3), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>                           <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a>                           <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a>                           <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>, </span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a>                           <span class="at">soft =</span> <span class="cn">FALSE</span>, <span class="at">thresh_len =</span> <span class="dv">20</span>, <span class="at">n_cv =</span> <span class="dv">10</span>, </span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a>                           <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a>res_linear4 <span class="ot">=</span> <span class="fu">secom_linear</span>(<span class="at">data =</span> <span class="fu">list</span>(tse4), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a>                           <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a>                           <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a>                           <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>), <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>, </span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a>                           <span class="at">soft =</span> <span class="cn">FALSE</span>, <span class="at">thresh_len =</span> <span class="dv">20</span>, <span class="at">n_cv =</span> <span class="dv">10</span>, </span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a>                           <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a>species1 <span class="ot">=</span> <span class="fu">rownames</span>(res_linear1<span class="sc">$</span>corr_fl)</span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a>species1 <span class="ot">=</span> species1[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species1)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species1)]</span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a>species2 <span class="ot">=</span> <span class="fu">rownames</span>(res_linear2<span class="sc">$</span>corr_fl)</span>
<span id="cb6-34"><a href="#cb6-34" tabindex="-1"></a>species2 <span class="ot">=</span> species2[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species2)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species2)]</span>
<span id="cb6-35"><a href="#cb6-35" tabindex="-1"></a>species3 <span class="ot">=</span> <span class="fu">rownames</span>(res_linear3<span class="sc">$</span>corr_fl)</span>
<span id="cb6-36"><a href="#cb6-36" tabindex="-1"></a>species3 <span class="ot">=</span> species3[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species3)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species3)]</span>
<span id="cb6-37"><a href="#cb6-37" tabindex="-1"></a>species4 <span class="ot">=</span> <span class="fu">rownames</span>(res_linear4<span class="sc">$</span>corr_fl)</span>
<span id="cb6-38"><a href="#cb6-38" tabindex="-1"></a>species4 <span class="ot">=</span> species4[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species4)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species4)]</span>
<span id="cb6-39"><a href="#cb6-39" tabindex="-1"></a>common_species <span class="ot">=</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(species1, species2, species3, species4))</span>
<span id="cb6-40"><a href="#cb6-40" tabindex="-1"></a>common_species <span class="ot">=</span> <span class="fu">sapply</span>(common_species, <span class="cf">function</span>(x) </span>
<span id="cb6-41"><a href="#cb6-41" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, x), </span>
<span id="cb6-42"><a href="#cb6-42" tabindex="-1"></a>                        <span class="fu">paste</span>(<span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">&quot;spp.&quot;</span>),</span>
<span id="cb6-43"><a href="#cb6-43" tabindex="-1"></a>                        <span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]))</span></code></pre></div>
</div>
<div id="visualization" class="section level2">
<h2>Visualization</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>df_corr1 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_linear1, <span class="at">type =</span> <span class="st">&quot;linear&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G1&quot;</span>)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>df_corr2 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_linear2, <span class="at">type =</span> <span class="st">&quot;linear&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G2&quot;</span>)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>df_corr3 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_linear3, <span class="at">type =</span> <span class="st">&quot;linear&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G3&quot;</span>)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>df_corr4 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_linear4, <span class="at">type =</span> <span class="st">&quot;linear&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G4&quot;</span>)</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>df_corr <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">&#39;rbind&#39;</span>, <span class="fu">list</span>(df_corr1, df_corr2, df_corr3, df_corr4)) <span class="sc">%&gt;%</span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;pair&quot;</span>, var1<span class="sc">:</span>var2, <span class="at">sep =</span> <span class="st">&quot; vs. &quot;</span>)</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>value_check <span class="ot">=</span> df_corr <span class="sc">%&gt;%</span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(pair) <span class="sc">%&gt;%</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">empty_idx =</span> <span class="fu">ifelse</span>(<span class="fu">all</span>(value <span class="sc">==</span> <span class="dv">0</span>), <span class="cn">TRUE</span>, <span class="cn">FALSE</span>))</span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a>non_empty_pair <span class="ot">=</span> value_check <span class="sc">%&gt;%</span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>  <span class="fu">filter</span>(empty_idx <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>  .<span class="sc">$</span>pair</span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>df_fig <span class="ot">=</span> df_corr <span class="sc">%&gt;%</span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pair <span class="sc">%in%</span> non_empty_pair)</span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a>fig_species_linear <span class="ot">=</span> df_fig <span class="sc">%&gt;%</span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a>  <span class="fu">ggline</span>(<span class="at">x =</span> <span class="st">&quot;group&quot;</span>, <span class="at">y =</span> <span class="st">&quot;value&quot;</span>,</span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">facet.by =</span> <span class="st">&quot;pair&quot;</span>,</span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Pearson Correlation&quot;</span>) <span class="sc">+</span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(value, <span class="dv">2</span>)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
</div>
<div id="nonlinear-correlations" class="section level1 tabset">
<h1 class="tabset">Nonlinear correlations</h1>
<div id="run-secom-1" class="section level2">
<h2>Run SECOM</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>res_dist1 <span class="ot">=</span> <span class="fu">secom_dist</span>(<span class="at">data =</span> <span class="fu">list</span>(tse1), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>                       <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>                       <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>                       <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>),</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>                       <span class="at">R =</span> <span class="dv">1000</span>, <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>res_dist2 <span class="ot">=</span> <span class="fu">secom_dist</span>(<span class="at">data =</span> <span class="fu">list</span>(tse2), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>                       <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>                       <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>                       <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>),</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>                       <span class="at">R =</span> <span class="dv">1000</span>, <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>res_dist3 <span class="ot">=</span> <span class="fu">secom_dist</span>(<span class="at">data =</span> <span class="fu">list</span>(tse3), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>                       <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>                       <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>                       <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>),</span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>                       <span class="at">R =</span> <span class="dv">1000</span>, <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb8-17"><a href="#cb8-17" tabindex="-1"></a>res_dist4 <span class="ot">=</span> <span class="fu">secom_dist</span>(<span class="at">data =</span> <span class="fu">list</span>(tse4), <span class="at">assay_name =</span> <span class="st">&quot;counts&quot;</span>,</span>
<span id="cb8-18"><a href="#cb8-18" tabindex="-1"></a>                       <span class="at">tax_level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">pseudo =</span> <span class="dv">0</span>, </span>
<span id="cb8-19"><a href="#cb8-19" tabindex="-1"></a>                       <span class="at">prv_cut =</span> <span class="fl">0.1</span>, <span class="at">lib_cut =</span> <span class="dv">1000</span>, <span class="at">corr_cut =</span> <span class="fl">0.5</span>, </span>
<span id="cb8-20"><a href="#cb8-20" tabindex="-1"></a>                       <span class="at">wins_quant =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.95</span>),</span>
<span id="cb8-21"><a href="#cb8-21" tabindex="-1"></a>                       <span class="at">R =</span> <span class="dv">1000</span>, <span class="at">thresh_hard =</span> <span class="fl">0.3</span>, <span class="at">max_p =</span> <span class="fl">0.005</span>, <span class="at">n_cl =</span> <span class="dv">2</span>)</span>
<span id="cb8-22"><a href="#cb8-22" tabindex="-1"></a></span>
<span id="cb8-23"><a href="#cb8-23" tabindex="-1"></a>species1 <span class="ot">=</span> <span class="fu">rownames</span>(res_dist1<span class="sc">$</span>corr_fl)</span>
<span id="cb8-24"><a href="#cb8-24" tabindex="-1"></a>species1 <span class="ot">=</span> species1[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species1)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species1)]</span>
<span id="cb8-25"><a href="#cb8-25" tabindex="-1"></a>species2 <span class="ot">=</span> <span class="fu">rownames</span>(res_dist2<span class="sc">$</span>corr_fl)</span>
<span id="cb8-26"><a href="#cb8-26" tabindex="-1"></a>species2 <span class="ot">=</span> species2[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species2)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species2)]</span>
<span id="cb8-27"><a href="#cb8-27" tabindex="-1"></a>species3 <span class="ot">=</span> <span class="fu">rownames</span>(res_dist3<span class="sc">$</span>corr_fl)</span>
<span id="cb8-28"><a href="#cb8-28" tabindex="-1"></a>species3 <span class="ot">=</span> species3[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species3)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species3)]</span>
<span id="cb8-29"><a href="#cb8-29" tabindex="-1"></a>species4 <span class="ot">=</span> <span class="fu">rownames</span>(res_dist4<span class="sc">$</span>corr_fl)</span>
<span id="cb8-30"><a href="#cb8-30" tabindex="-1"></a>species4 <span class="ot">=</span> species4[<span class="fu">grepl</span>(<span class="st">&quot;Species:&quot;</span>, species4)<span class="sc">|</span><span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, species4)]</span>
<span id="cb8-31"><a href="#cb8-31" tabindex="-1"></a>common_species <span class="ot">=</span> <span class="fu">Reduce</span>(intersect, <span class="fu">list</span>(species1, species2, species3, species4))</span>
<span id="cb8-32"><a href="#cb8-32" tabindex="-1"></a>common_species <span class="ot">=</span> <span class="fu">sapply</span>(common_species, <span class="cf">function</span>(x) </span>
<span id="cb8-33"><a href="#cb8-33" tabindex="-1"></a>  <span class="fu">ifelse</span>(<span class="fu">grepl</span>(<span class="st">&quot;Genus:&quot;</span>, x), </span>
<span id="cb8-34"><a href="#cb8-34" tabindex="-1"></a>                        <span class="fu">paste</span>(<span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">&quot;spp.&quot;</span>),</span>
<span id="cb8-35"><a href="#cb8-35" tabindex="-1"></a>                        <span class="fu">strsplit</span>(x, <span class="st">&quot;:&quot;</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]))</span></code></pre></div>
</div>
<div id="visualization-1" class="section level2">
<h2>Visualization</h2>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>df_corr1 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_dist1, <span class="at">type =</span> <span class="st">&quot;dist&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G1&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>df_corr2 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_dist2, <span class="at">type =</span> <span class="st">&quot;dist&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G2&quot;</span>)</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>df_corr3 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_dist3, <span class="at">type =</span> <span class="st">&quot;dist&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G3&quot;</span>)</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>df_corr4 <span class="ot">=</span> <span class="fu">data_preprocess</span>(res_dist4, <span class="at">type =</span> <span class="st">&quot;dist&quot;</span>, <span class="at">level =</span> <span class="st">&quot;Species&quot;</span>, <span class="at">tax =</span> key_species) <span class="sc">%&gt;%</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">group =</span> <span class="st">&quot;G4&quot;</span>)</span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" tabindex="-1"></a>df_corr <span class="ot">=</span> <span class="fu">do.call</span>(<span class="st">&#39;rbind&#39;</span>, <span class="fu">list</span>(df_corr1, df_corr2, df_corr3, df_corr4)) <span class="sc">%&gt;%</span></span>
<span id="cb9-11"><a href="#cb9-11" tabindex="-1"></a>  <span class="fu">unite</span>(<span class="st">&quot;pair&quot;</span>, var1<span class="sc">:</span>var2, <span class="at">sep =</span> <span class="st">&quot; vs. &quot;</span>)</span>
<span id="cb9-12"><a href="#cb9-12" tabindex="-1"></a></span>
<span id="cb9-13"><a href="#cb9-13" tabindex="-1"></a>value_check <span class="ot">=</span> df_corr <span class="sc">%&gt;%</span></span>
<span id="cb9-14"><a href="#cb9-14" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">group_by</span>(pair) <span class="sc">%&gt;%</span></span>
<span id="cb9-15"><a href="#cb9-15" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">empty_idx =</span> <span class="fu">ifelse</span>(<span class="fu">all</span>(value <span class="sc">==</span> <span class="dv">0</span>), <span class="cn">TRUE</span>, <span class="cn">FALSE</span>))</span>
<span id="cb9-16"><a href="#cb9-16" tabindex="-1"></a></span>
<span id="cb9-17"><a href="#cb9-17" tabindex="-1"></a>non_empty_pair <span class="ot">=</span> value_check <span class="sc">%&gt;%</span></span>
<span id="cb9-18"><a href="#cb9-18" tabindex="-1"></a>  <span class="fu">filter</span>(empty_idx <span class="sc">==</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb9-19"><a href="#cb9-19" tabindex="-1"></a>  .<span class="sc">$</span>pair</span>
<span id="cb9-20"><a href="#cb9-20" tabindex="-1"></a>df_fig <span class="ot">=</span> df_corr <span class="sc">%&gt;%</span></span>
<span id="cb9-21"><a href="#cb9-21" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pair <span class="sc">%in%</span> non_empty_pair)</span>
<span id="cb9-22"><a href="#cb9-22" tabindex="-1"></a></span>
<span id="cb9-23"><a href="#cb9-23" tabindex="-1"></a>fig_species_dist <span class="ot">=</span> df_fig <span class="sc">%&gt;%</span></span>
<span id="cb9-24"><a href="#cb9-24" tabindex="-1"></a>  <span class="fu">ggline</span>(<span class="at">x =</span> <span class="st">&quot;group&quot;</span>, <span class="at">y =</span> <span class="st">&quot;value&quot;</span>,</span>
<span id="cb9-25"><a href="#cb9-25" tabindex="-1"></a>         <span class="at">color =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">facet.by =</span> <span class="st">&quot;pair&quot;</span>,</span>
<span id="cb9-26"><a href="#cb9-26" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">&quot;&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;&quot;</span>, <span class="at">title =</span> <span class="st">&quot;Distance Correlation&quot;</span>) <span class="sc">+</span></span>
<span id="cb9-27"><a href="#cb9-27" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="fl">0.8</span>, <span class="fl">0.2</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.9</span>)) <span class="sc">+</span></span>
<span id="cb9-28"><a href="#cb9-28" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">round</span>(value, <span class="dv">2</span>)), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb9-29"><a href="#cb9-29" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span></code></pre></div>
</div>
</div>
<div id="outputs" class="section level1">
<h1>Outputs</h1>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">file =</span> <span class="st">&quot;../results/figures/supp_secom.pdf&quot;</span>, </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>       <span class="at">plot =</span> fig_species_linear, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>)</span></code></pre></div>
</div>
<div id="session-information" class="section level1">
<h1>Session information</h1>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.0

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] TreeSummarizedExperiment_2.10.0 GenomicRanges_1.54.1           
 [3] SummarizedExperiment_1.32.0     SingleCellExperiment_1.24.0    
 [5] IRanges_2.36.0                  S4Vectors_0.40.1               
 [7] DT_0.30                         ANCOMBC_2.4.0                  
 [9] microbiome_1.24.0               phyloseq_1.46.0                
[11] ggpubr_0.6.0                    tidygraph_1.3.0                
[13] ggraph_2.1.0                    igraph_1.5.1                   
[15] lubridate_1.9.3                 forcats_1.0.0                  
[17] stringr_1.5.0                   dplyr_1.1.4                    
[19] purrr_1.0.2                     tidyr_1.3.1                    
[21] tibble_3.2.1                    ggplot2_3.5.0                  
[23] tidyverse_2.0.0                 openxlsx_4.2.5.2               
[25] readr_2.1.4                    

loaded via a namespace (and not attached):
  [1] fs_1.6.3                    matrixStats_1.0.0          
  [3] bitops_1.0-7                DirichletMultinomial_1.44.0
  [5] httr_1.4.7                  doParallel_1.0.17          
  [7] numDeriv_2016.8-1.1         tools_4.3.1                
  [9] doRNG_1.8.6                 backports_1.4.1            
 [11] utf8_1.2.4                  R6_2.5.1                   
 [13] vegan_2.6-4                 lazyeval_0.2.2             
 [15] mgcv_1.8-42                 rhdf5filters_1.13.2        
 [17] permute_0.9-7               withr_2.5.1                
 [19] gridExtra_2.3               textshaping_0.3.7          
 [21] cli_3.6.2                   Biobase_2.62.0             
 [23] sandwich_3.0-2              sass_0.4.7                 
 [25] mvtnorm_1.2-3               proxy_0.4-27               
 [27] systemfonts_1.0.5           yulab.utils_0.1.0          
 [29] foreign_0.8-84              scater_1.30.0              
 [31] decontam_1.22.0             readxl_1.4.3               
 [33] rstudioapi_0.16.0           RSQLite_2.3.2              
 [35] generics_0.1.3              vroom_1.6.4                
 [37] gtools_3.9.4                car_3.1-2                  
 [39] zip_2.3.0                   Matrix_1.6-1.1             
 [41] biomformat_1.30.0           ggbeeswarm_0.7.2           
 [43] fansi_1.0.5                 DescTools_0.99.50          
 [45] DECIPHER_2.30.0             abind_1.4-5                
 [47] lifecycle_1.0.3             multcomp_1.4-25            
 [49] yaml_2.3.7                  carData_3.0-5              
 [51] rhdf5_2.46.0                SparseArray_1.2.2          
 [53] Rtsne_0.16                  grid_4.3.1                 
 [55] blob_1.2.4                  crayon_1.5.2               
 [57] lattice_0.21-8              beachmat_2.18.0            
 [59] pillar_1.9.0                knitr_1.45                 
 [61] boot_1.3-28.1               gld_2.6.6                  
 [63] codetools_0.2-19            glue_1.6.2                 
 [65] data.table_1.14.8           MultiAssayExperiment_1.28.0
 [67] vctrs_0.6.4                 treeio_1.26.0              
 [69] Rdpack_2.6                  cellranger_1.1.0           
 [71] gtable_0.3.4                cachem_1.0.8               
 [73] xfun_0.40                   rbibutils_2.2.16           
 [75] S4Arrays_1.2.0              survival_3.5-5             
 [77] iterators_1.0.14            bluster_1.12.0             
 [79] gmp_0.7-2                   TH.data_1.1-2              
 [81] nlme_3.1-162                bit64_4.0.5                
 [83] GenomeInfoDb_1.38.0         bslib_0.5.1                
 [85] irlba_2.3.5.1               vipor_0.4.5                
 [87] rpart_4.1.19                colorspace_2.1-0           
 [89] BiocGenerics_0.48.0         DBI_1.1.3                  
 [91] Hmisc_5.1-1                 nnet_7.3-19                
 [93] ade4_1.7-22                 Exact_3.2                  
 [95] tidyselect_1.2.0            bit_4.0.5                  
 [97] compiler_4.3.1              htmlTable_2.4.2            
 [99] BiocNeighbors_1.20.0        expm_0.999-7               
[101] DelayedArray_0.28.0         checkmate_2.3.0            
[103] scales_1.3.0                digest_0.6.33              
[105] minqa_1.2.6                 rmarkdown_2.25             
[107] XVector_0.42.0              htmltools_0.5.6.1          
[109] pkgconfig_2.0.3             base64enc_0.1-3            
[111] lme4_1.1-34                 sparseMatrixStats_1.14.0   
[113] MatrixGenerics_1.14.0       fastmap_1.1.1              
[115] rlang_1.1.3                 htmlwidgets_1.6.2          
[117] DelayedMatrixStats_1.24.0   farver_2.1.1               
[119] jquerylib_0.1.4             zoo_1.8-12                 
[121] jsonlite_1.8.7              energy_1.7-11              
[123] BiocParallel_1.36.0         BiocSingular_1.18.0        
[125] RCurl_1.98-1.12             magrittr_2.0.3             
[127] Formula_1.2-5               scuttle_1.12.0             
[129] GenomeInfoDbData_1.2.11     Rhdf5lib_1.24.0            
[131] munsell_0.5.0               Rcpp_1.0.12                
[133] ape_5.7-1                   viridis_0.6.4              
[135] CVXR_1.0-11                 stringi_1.7.12             
[137] rootSolve_1.8.2.4           zlibbioc_1.48.0            
[139] MASS_7.3-60                 plyr_1.8.9                 
[141] parallel_4.3.1              ggrepel_0.9.4              
[143] lmom_3.0                    Biostrings_2.70.1          
[145] graphlayouts_1.1.0          splines_4.3.1              
[147] multtest_2.58.0             hms_1.1.3                  
[149] ggsignif_0.6.4              rngtools_1.5.2             
[151] reshape2_1.4.4              stats4_4.3.1               
[153] ScaledMatrix_1.10.0         evaluate_0.22              
[155] nloptr_2.0.3                tzdb_0.4.0                 
[157] foreach_1.5.2               tweenr_2.0.2               
[159] polyclip_1.10-6             ggforce_0.4.1              
[161] rsvd_1.0.5                  broom_1.0.5                
[163] Rmpfr_0.9-3                 e1071_1.7-13               
[165] tidytree_0.4.5              rstatix_0.7.2              
[167] ragg_1.2.6                  viridisLite_0.4.2          
[169] class_7.3-22                gsl_2.1-8                  
[171] lmerTest_3.1-3              memoise_2.0.1              
[173] beeswarm_0.4.0              cluster_2.1.4              
[175] timechange_0.2.0            mia_1.10.0                 </code></pre>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
